widget/editor.go

```diff
@@ -318,22 +445,64 @@ func (e *Editor) command(gtx layout.Context, k key.Event) bool {
 		e.moveStart()
 	case key.NameEnd:
 		e.moveEnd()
+	// Initiate a paste operation, by requesting the clipboard contents; other
+	// half is in Editor.processKey() under clipboard.Event.
 	case "V":
 		if k.Modifiers != key.ModShortcut {
 			return false
 		}
 		clipboard.ReadOp{Tag: &e.eventKey}.Add(gtx.Ops)
-	case "C":
+	// Copy or Cut selection -- ignored if nothing selected.
+	case "C", "X":
+		if k.Modifiers != key.ModShortcut || !e.IsSelected() {
+			return false
+		}
+		clipboard.WriteOp{Text: e.SelectedText()}.Add(gtx.Ops)
+		if k.Name == "X" {
+			e.DeleteSelection()
+		}
+	// Select all
+	case "A":
 		if k.Modifiers != key.ModShortcut {
 			return false
 		}
-		clipboard.WriteOp{Text: e.Text()}.Add(gtx.Ops)
+		s.Start.ofs = 0
+		s.End.ofs = e.Len()
+		s.makeValid(e)
 	default:
 		return false
 	}
+
+	// Process "shifted" movement, e.g. shift-UpArrow.
+	if mt := movementType(k); mt > 0 {
+		if k.Modifiers.Contain(key.ModShift) {
+			if e.IsSelected() {
```

Comment:
Again, I wouldn't expect an IsSelected check here because the behaviour is the same regardless of the selection: shift+arrow moves the selection end, period.

Reply:
No.  See my response to your comment on line 273.

Reply:
Maybe I'm being dense, but judging from the GH comment editor in macOS, shift+arrow really only move the selection end regardless of where the end was to begin with. That is, assuming selectionEnd = selectionStart = caretPos for the normal case of nothing selected.

Reply:
Ah.  For the normal case of nothing selected, that is true, I set the selection start to where the cursor was and the selection end to where the cursor is.  For the case where you do have something selected, you should move the last side that you moved (`*e.dragSide`), which (given the rest of the code as it is now) could be the front of the selection.  Consider if you have nothing selected, type shift-down-arrow, and then shift-click before the start of the selection, and then say shift-up-arrow (this video shows more than just that):

https://user-images.githubusercontent.com/2324697/103782559-9cc2c200-5005-11eb-983d-9f8fe978bf63.mov

Reply:
Indeed, this seems like another Chrome/Firefox difference. Please keep your preferred behaviour.